swayfmt/src/items/item_fn/mod.rs

```diff
@@ -51,11 +51,11 @@ impl Format for ItemFn {
 
                     Self::close_curly_brace(formatted_code, formatter)?;
                 } else {
+                    let range = self.span().into();
                     Self::open_curly_brace(formatted_code, formatter)?;
-                    formatter.indent();
-                    let comments = write_comments(formatted_code, self.span().into(), formatter)?;
-                    if !comments {
-                        formatter.unindent();
+                    if has_comments_in_formatter(formatter, &range) {
+                        formatter.indent();
+                        write_comments(formatted_code, range, formatter)?;
```

Comment:
Does this change the behavior of this block or just make it more readable?

I believe this pattern of checking `!comments` is used in a bunch of other parts of the formatter. We might want to use it everywhere to establish it as a pattern.

Reply:
If we check it after we would have to remove the bits from the strings (the newline that was added). If we checked this way we prevent adding it, this kind of things was introduced in #5148 (which is nicer I think)